### PR TITLE
Release 3.4.0

### PR DESCRIPTION
**Pull request by nicoddemus**
Builds:

Travis: https://travis-ci.org/nicoddemus/devpi-cloud-test-pytest/builds/334873331
AppVeyor: https://ci.appveyor.com/project/nicoddemus/devpi-cloud-test-pytest/build/1.0.34

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -8,6 +8,135 @@
 
 .. towncrier release notes start
 
+Pytest 3.4.0 (2018-01-29)
+=========================
+
+Deprecations and Removals
+-------------------------
+
+- All pytest classes now subclass ``object`` for better Python 3 compatibility.
```
Comment (blueyed) [inline CHANGELOG.rst]: s/2/3 ?!

File: CHANGELOG.rst [hunk 2]
```diff
@@ -8,6 +8,135 @@
...
+  less error prone. (`#2527
+  <https://github.com/pytest-dev/pytest/issues/2527>`_)
+
+- **Incompatible change**: after community feedback the `logging
+  <https://docs.pytest.org/en/latest/logging.html>`_ functionality has
+  undergone some changes. Please consult the `logging documentation
+  <https://docs.pytest.org/en/latest/logging.html#incompatible-changes-in-pytest-3-4>`_
+  for details. (`#3013 <https://github.com/pytest-dev/pytest/issues/3013>`_)
+
+- Console output fallsback to "classic" mode when capture is disabled (``-s``),
```
Comment (blueyed) [inline CHANGELOG.rst]: s/fallsback/falls back/

Comment (blueyed) [reply CHANGELOG.rst]: s/capture/capturing/

File: CHANGELOG.rst [hunk 3]
```diff
@@ -8,6 +8,135 @@
...
+  <https://docs.pytest.org/en/latest/writing_plugins.html#_pytest.hookspec.pytest_runtest_logfinish>`_
+  hook which is called when a test item has finished executing, analogous to
+  `pytest_runtest_logstart
+  <https://docs.pytest.org/en/latest/writing_plugins.html#_pytest.hookspec.pytest_runtest_start>`_.
+  (`#3101 <https://github.com/pytest-dev/pytest/issues/3101>`_)
+
+- Improve performance when collecting tests using many fixtures. (`#3107
+  <https://github.com/pytest-dev/pytest/issues/3107>`_)
+
+- New ``caplog.get_records(when)`` method which provides access the captured
```
Comment (blueyed) [inline CHANGELOG.rst]: s/access the/access to the/

File: CHANGELOG.rst (hunk 2 above)
Comment (The-Compiler) [inline CHANGELOG.rst]: fallsback -> falls back

File: CHANGELOG.rst [hunk 4]
```diff
@@ -8,6 +8,135 @@
...
+  records during each testing stage: ``"setup"``, ``"call"`` and ``"teardown"``
```
Comment (blueyed) [inline CHANGELOG.rst]: s/:/for the/

File: CHANGELOG.rst [hunk 5]
```diff
@@ -8,6 +8,135 @@
...
+- The default cache directory has been renamed from ``.cache`` to
+  ``.pytest_cache`` after community feedback that the name ``.cache`` did not
+  make it clear that it was used by pytest. (`#3138
+  <https://github.com/pytest-dev/pytest/issues/3138>`_)
+
+
+Bug Fixes
+---------
+
+- Fixed hanging pexpect test on MacOS by using flush() instead of wait().
```
Comment (blueyed) [inline CHANGELOG.rst]: s/Fixed/Fix - also below.

File: CHANGELOG.rst [hunk 6]
```diff
@@ -8,6 +8,135 @@
...
+  (`#2022 <https://github.com/pytest-dev/pytest/issues/2022>`_)
+
+- Fixed restoring Python state after in-process pytest runs with the
+  ``pytester`` plugin; this may break tests using making multiple inprocess
```
Comment (blueyed) [inline CHANGELOG.rst]: s/making // ?!

File: CHANGELOG.rst [hunk 7]
```diff
@@ -8,6 +8,135 @@
...
+  <https://github.com/pytest-dev/pytest/issues/2423>`_)
+
+- Clarify that warning capturing doesn't change the warning filter by default.
+  (`#2457 <https://github.com/pytest-dev/pytest/issues/2457>`_)
+
+- Clarify a possible confusion when using pytest_fixture_setup with fixture
+  functions that return None. (`#2698
+  <https://github.com/pytest-dev/pytest/issues/2698>`_)
+
+- Fix the wording of a sentence on doctest flags use in pytest. (`#3076
```
Comment (blueyed) [inline CHANGELOG.rst]: s/use/used/

File: CHANGELOG.rst (hunk 1 above)
Comment (nicoddemus) [reply CHANGELOG.rst]: I meant "compatibility with Python 3", but I think changing to "Python 2/3 compatibility will be better" 👍

File: CHANGELOG.rst (hunk 2 above)
Comment (nicoddemus) [reply CHANGELOG.rst]: Done

Comment (nicoddemus) [reply CHANGELOG.rst]: Done

File: CHANGELOG.rst (hunk 3 above)
Comment (nicoddemus) [reply CHANGELOG.rst]: Done

File: CHANGELOG.rst (hunk 4 above)
Comment (nicoddemus) [reply CHANGELOG.rst]: Done

File: CHANGELOG.rst (hunk 5 above)
Comment (nicoddemus) [reply CHANGELOG.rst]: done

File: CHANGELOG.rst (hunk 6 above)
Comment (nicoddemus) [reply CHANGELOG.rst]: done

File: CHANGELOG.rst (hunk 7 above)
Comment (nicoddemus) [reply CHANGELOG.rst]: done

**Comment by nicoddemus** [comment]
Uploaded a new package now with #3142 

Travis: https://travis-ci.org/nicoddemus/devpi-cloud-test-pytest/builds/335315284
AppVeyor: https://ci.appveyor.com/project/nicoddemus/devpi-cloud-test-pytest/build/1.0.35

**Pull request merged**